src/base/memory_hash.h

```diff
@@ -76,15 +77,15 @@ struct mht_table
 
 extern unsigned int mht_2str_pseudo_key (const void *key, int key_size);
 extern unsigned int mht_1strlowerhash (const void *key, const unsigned int ht_size);
-extern unsigned int mht_1strhash (const void *key, const unsigned int ht_size);
+extern unsigned int mht_1strhash (const void *key, const unsigned int ht_size, unsigned int *orig_hash_value);
 extern unsigned int mht_2strhash (const void *key, const unsigned int ht_size);
 extern unsigned int mht_3strhash (const void *key, const unsigned int ht_size);
-extern unsigned int mht_4strhash (const void *key, const unsigned int ht_size);
-extern unsigned int mht_5strhash (const void *key, const unsigned int ht_size);
+extern unsigned int mht_4strhash (const void *key, const unsigned int ht_size, unsigned int *orig_hash_value);
+extern unsigned int mht_5strhash (const void *key, const unsigned int ht_size, unsigned int *orig_hash_value);
 extern unsigned int mht_numhash (const void *key, const unsigned int ht_size);
 
 extern unsigned int mht_get_hash_number (const int unsigned ht_size, const DB_VALUE * val);
-extern unsigned int mht_ptrhash (const void *ptr, const unsigned int ht_size);
+extern unsigned int mht_ptrhash (const void *ptr, const unsigned int ht_size, unsigned int *orig_hash_value);
```

Comment:
If the value type is an int, wouldn't the comparison of orig_hash_value be inefficient?

Reply:
It is inevitable to fit the common function prototype.
In comparison, one comparison is increased only when an actual match is made.